moved slides files into slides directory

butterflyology opened the pull request:
Moved the `.Rmd` and `.html` files into the `slides` directory. 

https://butterflyology.github.io/rr-publication

tracykteal commented:
It looks like 01-publication-slides.html and publication-01-slides.html are redundant. It might be best to re-render them from the current Rmd and commit an updated HTML file.

butterflyology commented:
There were two slide decks that contained the same information, just formatted differently. I removed the "older" slide deck and renamed the "current" slide deck to follow the agreed upon nomenclature.
